### PR TITLE
Bookstore : Add redux with react

### PR DESCRIPTION
**Pull request by ehmaddd**
in this PR, the following features are added :
-  Wrapped the App with the Provider component from react-redux
-  Passed the configured store into the Provider
-  Createed the initialState variable for the booksSlice with initalstate
-  Display your books, received from the slice, in a reusable component
-  Dispatched actions using useDispatch
-  Added a Button component, which includes an event handler that adds a book to the books array (with attributes id, title and author)
-  Added a Button component, which includes an event handler that removes a book from the books array (by id)
- There is no linter errors
-  Used correct Gitflow
-  Documented the work in a professional way
-  Followed best practices for HTML, CSS and Javascript

### CODE REVIEW

File: src/components/AddBook.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  const authorField = document.querySelector('#author');
+  const titleText = titleField.value;
+  const authorText = authorField.value;
+  titleField.value = '';
+  authorField.value = '';
+  return {
+    title: titleText,
+    author: authorText,
+  };
+};
```
Comment (didierganthier) [inline src/components/AddBook.js]: Well done overall on the project 💯 but here you are directly interacting with the `DOM` which you shouldn't do in React, instead your component should have a local state for `title` and `author`, which should be updated through the `onChange` event. Consider using the `useState` hook:

```jsx
  const [title, setTitle] = useState("")
  const [author, setAuthor] = useState("")
```

**Review by didierganthier — CHANGES_REQUESTED** [review]
Hi @ehmaddd ,

🎉 Good job so far! You're making great progress on this project, but there are still a few areas that need attention before we can move on to the next step.



## Highlights
- `initialState` has been added to the `booksSlice` ✔️ 
- The Redux `store` has been passed into the `<Provider>` ✔️ 
- `books` are accessed from inside the `<Books>` component with `useSelector` ✔️ 

## Required Changes 🛠
- Please review the comments under the review for specific changes that still need to be made.

## Optional Suggestions 💡
- While these comments with the [OPTIONAL] prefix won't prevent the approval of this pull request, I strongly recommend taking them into account as they can improve your code.

🚀 Keep pushing forward, you're almost there! If you have any questions or need clarification on anything, feel free to leave a comment in the pull request thread and don't forget to tag me so I'll receive the notification.

**Please note**
- To avoid confusion and delays, please do not open a new pull request for re-reviews. Use the same pull request submitted for the first review, unless otherwise instructed.

------
As per the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project. If you feel that the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s). 

Cheers and happy coding! 👏👏👏

**Review by didierganthier — CHANGES_REQUESTED** [review]
Hi @ehmaddd ,

🎉 Good job so far! You're making great progress on this project, but there are still a few areas that need attention before we can move on to the next step.



## Highlights
- `initialState` has been added to the `booksSlice` ✔️ 
- The Redux `store` has been passed into the `<Provider>` ✔️ 
- `books` are accessed from inside the `<Books>` component with `useSelector` ✔️ 

## Required Changes 🛠
- Please review the comments under the review for specific changes that still need to be made.

## Optional Suggestions 💡
- While these comments with the [OPTIONAL] prefix won't prevent the approval of this pull request, I strongly recommend taking them into account as they can improve your code.

🚀 Keep pushing forward, you're almost there! If you have any questions or need clarification on anything, feel free to leave a comment in the pull request thread and don't forget to tag me so I'll receive the notification.

**Please note**
- To avoid confusion and delays, please do not open a new pull request for re-reviews. Use the same pull request submitted for the first review, unless otherwise instructed.

------
As per the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project. If you feel that the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s). 

Cheers and happy coding! 👏👏👏

File: src/components/AddBook.js (hunk 1 above)
Comment (ehmaddd) [reply src/components/AddBook.js]: I have solved all the issues mentioned as per instructed. Thanks for the kind suggestion

**Review by Shisui6 — APPROVED** [review]
Hi @ehmaddd, 

# Approved 🪄
Your project is complete! There is nothing else to say other than... it's time to merge it 💎
Congratulations! 🎉 

## Highlights 🎯
- Add and remove buttons work ✅


## Optional suggestions 🪶

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer, and addressing them will really improve your application._

You can also consider the following:

N/A

Cheers and Happy coding!🍻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**